protobuf/mapping.go

```diff
@@ -17,12 +37,35 @@ func (t *ProtoType) Type() Type {
 	return NewNamed(t.Package, t.Name)
 }
 
+type Decorator func(*Package, *Message, *Field)
+type Decorators []Decorator
+
+func NewDecorators(fns ...func(*Package, *Message, *Field)) Decorators {
+	var decorators Decorators
+
+	for _, fn := range fns {
+		decorators = append(decorators, Decorator(fn))
+	}
+
+	return decorators
+}
+
+func (d Decorator) Run(p *Package, m *Message, f *Field) {
```

Comment:
Had problems with type declarations to functions yesterday. I also would like to improve this such that _defensive programming_ is not needed in the decorators. If a `decorator` is run via `Run`, we have more space to improve our coding experience later.